File: commons/javafx/src/main/java/com/cedarsoft/commons/javafx/FloatTextFormatterFilter.java
package com.cedarsoft.commons.javafx;

import java.util.function.Predicate;

import javax.annotation.Nonnull;

import javafx.util.converter.NumberStringConverter;

/**
 * Processes text formatter changes. Ensures only valid floats are added
 */
public class FloatTextFormatterFilter extends NumberTextFormatterFilter {

  public FloatTextFormatterFilter(@Nonnull NumberStringConverter converter, @Nonnull Predicate<Float> filter) {
    super(converter, number -> filter.test(number.floatValue()), 0.0);
  }

}

File: commons/javafx/src/main/java/com/cedarsoft/commons/javafx/FloatTextFormatterFilter.kt
package com.cedarsoft.commons.javafx

import javafx.util.converter.NumberStringConverter
import java.util.function.Predicate

/**
 * Processes text formatter changes. Ensures only valid floats are added
 */
class FloatTextFormatterFilter(
  converter: NumberStringConverter,
  filter: Predicate<Float>,
) : NumberTextFormatterFilter(
  converter = converter,
  filter = Predicate { number: Number -> filter.test(number.toFloat()) },
  emptyValue = 0.0
)
